src/Store/VimStoreConnector.re

```diff
@@ -975,6 +980,42 @@ let start =
     });
   };
 
+  let zoomStep = 0.2;
+  let defaultZoom = 1.0;
+  let minZoomValue = 0.0;
+  let maxZoomValue = 2.8;
+  let zoomEffect = (state: State.t, zoomAction) =>
+    Isolinear.Effect.create(~name="vim.zoom", () => {
+      let configuration = state.configuration;
+      let currentZoomValue =
+        Core.Configuration.getValue(c => c.uiZoom, configuration);
+
+      let calculatedZoomValue =
+        switch (zoomAction) {
+        | ZoomIn => currentZoomValue +. zoomStep
+        | ZoomOut => currentZoomValue -. zoomStep
+        | ZoomReset => defaultZoom
+        };
+
+      let newZoomValue =
+        calculatedZoomValue < minZoomValue
+          ? minZoomValue
+          : calculatedZoomValue > maxZoomValue
+              ? maxZoomValue : calculatedZoomValue;
```

Comment:
You can use `Utility.IntEx.clamp` for this. It would be significantly more readable I think.